### PR TITLE
feat: added reorderable list and updated fieldProps

### PR DESCRIPTION
**Pull request by Tormak9970**
Added the reorderable list component and added `style` property to the `FieldProps` type.

Example of how to implement the reorderable list can be found [here](https://github.com/Tormak9970/bash-shortcuts/blob/3be7c232c64158636dfb8f0f4550086def873394/src/components/shortcuts-manager/ManageShortcuts.tsx#L14)

### CODE REVIEW

File: src/custom-components/ReorderableList.tsx
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:268]: Class name is a bit too generic. Also why not just use an inline style here.

File: src/custom-components/ReorderableList.tsx
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:112]: Class name too generic

**Review by AAGaming00 — DISMISSED** [review]

File: src/custom-components/ReorderableList.tsx
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx:63]: From what I understand that is the intended use of useref. Also why convert the to arrow functions?

I'll rename the classes and inline the scoper style when I have a chance.

**Comment by Tormak9970** [comment]
Working on a new Toggle version that's simpler and more useful. Will update the PR when its done and tested.

**Comment by Tormak9970** [comment]
@AAGaming00 Would appreciate you taking a look in case you want me to make changes.

File: src/custom-components/ReorderableList.tsx
Comment (EMERALD0874) [inline src/custom-components/ReorderableList.tsx:50]: Is this style necessary?

File: src/custom-components/ReorderableList.tsx
Comment (EMERALD0874) [inline src/custom-components/ReorderableList.tsx:80]: You may want to have a Decky maintainer run this through a linter so it matches our Prettier config.

File: package.json
Comment (EMERALD0874) [inline package.json:83]: react-icons is not used in this PR. Please revert changes to package.json and pnpm-lock.yaml as this was the only change made to both.

**Review by EMERALD0874 — CHANGES_REQUESTED** [review]
Please see the attached comments.

**Review by EMERALD0874 — APPROVED** [review]
Looks good! Just please make sure a Decky maintainer runs this through Prettier before merge.

File: src/custom-components/ReorderableList.tsx
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:20]: use `@example` for better doc

File: src/custom-components/ReorderableList.tsx
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:4]: add documentation for these

File: src/custom-components/ReorderableList.tsx
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:72]: see first comment

File: src/custom-components/ReorderableList.tsx
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:41]: inline style probably better here, we don't want dfl to mess with global CSS too much

File: src/custom-components/ReorderableList.tsx
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:120]: why the `@ts-ignore`?

**Review by AAGaming00 — DISMISSED** [review]

**Comment by AAGaming00** [comment]
> Looks good! Just please make sure a Decky maintainer runs this through Prettier before merge.

Seems like maintainer access isn't working as Tormack didn't create a new branch on their fork for this PR.

```
❯ git push
remote: Permission to Tormak9970/decky-frontend-lib.git denied to AAGaming00.
fatal: unable to access 'https://github.com/Tormak9970/decky-frontend-lib/': The requested URL returned error: 403
```

**Comment by Tormak9970** [comment]
Will make the requested changes and run prettier myself

File: src/custom-components/ReorderableList.tsx
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx:20]: You want me to put the example in the jsDoc comment?

File: src/custom-components/ReorderableList.tsx
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx:120]: fixed by specifying type on css obj

File: src/custom-components/ReorderableList.tsx
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx]: ran prettier on newest update

**Review by TrainDoctor — APPROVED** [review]

**Review by AAGaming00 — APPROVED** [review]

**Pull request merged**
